src/runtime/actor/context.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    any::Any,
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+    panic::AssertUnwindSafe,
+};
+
+use futures::{
+    future::{AbortRegistration, Abortable, Aborted},
+    FutureExt, Stream,
+};
+
+use super::{
+    envelope::{Envelope, HandleEvent},
+    handle::Addr,
+    report::Report,
+    Actor,
+};
+use crate::runtime::{config::SpawnConfig, scope::RuntimeScope, shutdown::ShutdownStream};
+
+/// The context that an actor can use to interact with the runtime
+pub struct ActorContext<A: Actor> {
+    pub(crate) scope: RuntimeScope,
+    pub(crate) handle: Addr<A>,
+    pub(crate) receiver: ShutdownStream<Box<dyn Stream<Item = Envelope<A>> + Unpin + Send>>,
```

Comment:
consider introducing the typedef
`type EnvelopeStream = Box<dyn Stream<Item = Envelope<A>> + Unpin + Send`;
because it reduces refactoring this definition to a one-line change.